src/main/java/com/czertainly/core/model/request/CrmfCertificateRequest.java

```diff
@@ -0,0 +1,53 @@
+package com.czertainly.core.model.request;
+
+import com.czertainly.core.util.CertificateUtil;
+import org.bouncycastle.asn1.crmf.CRMFObjectIdentifiers;
+import org.bouncycastle.asn1.x500.X500Name;
+import org.bouncycastle.asn1.x509.*;
+import org.bouncycastle.cert.crmf.CertificateRequestMessage;
+import org.bouncycastle.openssl.PEMException;
+import org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter;
+
+import java.io.IOException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PublicKey;
+import java.util.Map;
+
+public class CrmfCertificateRequest implements CertificateRequest {
+
+    private final byte[] encoded;
+    private final CertificateRequestMessage certificateRequestMessage;
```

Comment:
Should this be `JcaCertificateRequestMessage`? Similar like we are using `JcaPKCS10CertificateRequest`?